### PR TITLE
feat(module:affix&anchor&back-top&avatar): add components to library

### PR DESCRIPTION
**Pull request by cipchk**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added
- [x] Docs have been added

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: src/components/anchor/nz-anchor-link.component.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+  constructor(public el: ElementRef, private _anchorComp: NzAnchorComponent) {
+    this._anchorComp.add(this);
+  }
+
+  goToClick(e: Event) {
+    e.preventDefault();
+    e.stopPropagation();
+    this._anchorComp.scrollTo(this);
+    return false;
```
Comment (wilsoncook) [inline src/components/anchor/nz-anchor-link.component.ts]: Suggest: no need to "return false"

File: src/components/anchor/nz-anchor.component.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    this._visible = !!sections.length;
+    if (!this._visible)
+      return;
+
+    const maxSection = sections.reduce((prev, curr) => curr.top > prev.top ? curr : prev);
+    maxSection.comp.active = true;
+
+    let linkNode = (maxSection.comp.el.nativeElement as HTMLDivElement).querySelector('.ant-anchor-link-title') as HTMLElement;
+    this.ball.nativeElement.style.top = `${linkNode.offsetTop + linkNode.clientHeight / 2 - 4.5}px`;
+    console.log(linkNode, linkNode.offsetTop + linkNode.clientHeight / 2 - 4.5);
```
Comment (wilsoncook) [inline src/components/anchor/nz-anchor.component.ts]: Should remove debug info.

**Comment by LinBoLen** [comment]
it seems that you are use `CRLF`, please use `LF` Line Separator

**Comment by wilsoncook** [comment]
@LinBoLen Have noticed, but CRLF is default in Windows. Any side effects is good to be heard.

**Comment by mango-csl** [comment]
”树“控件计划什么时候引入进来

**Pull request merged**